### PR TITLE
[bugfix] filer: chunk mis-deletion

### PR DESCRIPTION
**Pull request by banjiaojuhao**
In file modification, old chunks will be mis-deleted when they are merged(Manifestized).

### CODE REVIEW

File: weed/filer/filer_deletion.go [hunk 1]
```diff
@@ -141,14 +147,36 @@ func (f *Filer) deleteChunksIfNotNew(oldEntry, newEntry *Entry) {
 
 	var toDelete []*filer_pb.FileChunk
 	newChunkIds := make(map[string]bool)
-	for _, newChunk := range newEntry.Chunks {
+	newDChunks, newMChunks, err := ResolveChunkManifest(f.MasterClient.GetLookupFileIdFunction(),
```
Comment (chrislusf) [inline weed/filer/filer_deletion.go]: please spell out the variable names.  saving a few characters in `newDChunks newMChunks` is not worth it.

Comment (banjiaojuhao) [reply weed/filer/filer_deletion.go]: I just followed the convention like:
https://github.com/chrislusf/seaweedfs/blob/da3d330616663e1c843514bd7a7a2c1833bf42d1/weed/filer/filechunk_manifest.go#L71
https://github.com/chrislusf/seaweedfs/blob/da3d330616663e1c843514bd7a7a2c1833bf42d1/weed/shell/command_volume_fsck.go#L156
https://github.com/chrislusf/seaweedfs/blob/da3d330616663e1c843514bd7a7a2c1833bf42d1/weed/shell/command_volume_fsck.go#L335

Should I change these `dChunks` to `dataChunks` together? @chrislusf

Comment (chrislusf) [reply weed/filer/filer_deletion.go]: yes, please.

Comment (banjiaojuhao) [reply weed/filer/filer_deletion.go]: done @chrislusf

**Pull request merged**